test/lib/testMayaUsdProxyAccessor.py

```diff
@@ -0,0 +1,1265 @@
+        
+        # Get UFE items
+        ufeItemSphere = makeUfePath(nodeDagPath,'/ParentA/Sphere')
+        worldMatrixPlugSphere = pa.GetOrCreateAccessPlug(ufeItemSphere, '', Sdf.ValueTypeNames.Matrix4d )
+        
+        cachingScope.waitForCache(self)
+        cachingScope.checkValidFrames(self, self.cache_allFrames)
+        
+        # Validate that DAG transformation is applied when reading world space matrix from USD
+        cmds.currentTime(1)
+        v1 = cmds.getAttr('{}.{}'.format(nodeDagPath,worldMatrixPlugSphere))
+        self.assertEqual(v1, [1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 5.0, 0.0, 0.0, 1.0])
+        
+        cmds.currentTime(100)
+        v1 = cmds.getAttr('{}.{}'.format(nodeDagPath,worldMatrixPlugSphere))
+        self.assertVectorAlmostEqual(v1, [0.0, 0.0, -1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 5.0, 5.0, 1.0])
+
+    def validateInput(self, cachingScope):
+        """
+        Validate that accessor can write data to the stage and it's propagated correctly to:
+        - output plugs
+        - UFE interfaces
+        """
+        nodeDagPath, stage = createProxyFromFile(self.testAnimatedHierarchyUsdFile)
+        
+        # Get UFE items
+        ufeItemParent = makeUfePath(nodeDagPath,'/ParentA')
+        ufeItemSphere = makeUfePath(nodeDagPath,'/ParentA/Sphere')
+        ufeItemCube = makeUfePath(nodeDagPath,'/ParentA/Cube')
+        
+        # Create accessor plugs
+        translatePlugParent = pa.GetOrCreateAccessPlug(ufeItemParent, usdAttrName='xformOp:translate')
+        rotatePlugParent = pa.GetOrCreateAccessPlug(ufeItemParent, usdAttrName='xformOp:rotateXYZ')
+        worldMatrixPlugSphere = pa.GetOrCreateAccessPlug(ufeItemSphere, '', Sdf.ValueTypeNames.Matrix4d)
+        
+        # Instanciate interface object for UFE transform3d
+        ufeTransform3dCube = ufe.Transform3d.transform3d(ufeItemCube)
+        
+        # Animate accessor plugs (they will become inputs to the stage owned by proxy)
+        cmds.setKeyframe( '{}.{}'.format(nodeDagPath,translatePlugParent), time=1.0, value=0.0 )
+        cmds.setKeyframe( '{}.{}'.format(nodeDagPath,translatePlugParent), time=100.0, value=10.0)
+        
+        cmds.setKeyframe( '{}.{}'.format(nodeDagPath,rotatePlugParent), time=1.0, value=0.0 )
+        cmds.setKeyframe( '{}.{}'.format(nodeDagPath,rotatePlugParent), time=100.0, value=90.0)
+        
+        # Validate that input accessor plugs wrote the data to USD
+        # Validate that output accessor plug has data driven by input accessor plug
+        
+        cachingScope.checkValidFrames(self, self.cache_empty)
+        cachingScope.waitForCache(self)
+        cachingScope.checkValidFrames(self, self.cache_allFrames)
+        
+        cmds.currentTime(1)
+        v1 = cmds.getAttr('{}.{}'.format(nodeDagPath,worldMatrixPlugSphere))
+        v2 = ufeTransform3dCube.segmentInclusiveMatrix()
```

Comment:
We are using segmentInclusiveMatrix() to get the matrix of the final segment of the UFE path, correct?  I.e. in this case, the USD matrix.

Reply:
Yes